routes/relationships.js

```diff
@@ -0,0 +1,252 @@
+        returnList.push(...mentorList);
+        
+        let returnObjects = []
+        for (let relationshipId of returnList){
+            returnObjects.push(await relationships.getRelationshipById(relationshipId));
+        };
+
+        let jsonObj = {
+            return: returnObjects,
+            success: true
+        };
+
+        // TODO: Not set on any of these layouts/frames being rendered since we're not yet sure exactly where these routes will be used
+       // res.render('frames/relationships', {layout: 'profile', relationships: returnList}); 
+       res.status(200).json(jsonObj);
+    }
+    catch(e) {
+        //res.status(500).render('frames/error', {layout: 'profile', error: "Internal server error"});
+        res.status(500).json({error:e});
+    }
+};
+
+/**
+ * Post a new relationship object 
+ * @param {Object} req - request object
+ * @param {Object} res - response object
+ * @returns void
+ */
+async function postNewRelationship(req, res){
+    let menteeId, mentorId, userId;
+    try {
+        userId = validate.convertID(req.params.userId);
+       // validate.checkArgLength(req.body, 4);
+        validate.checkIsEmptyString(req.body.relationshipDescription);
+        mentorId = validate.convertID(req.body.mentorId); 
+        menteeId = validate.convertID(req.body.menteeId);
+        validate.checkIsEmptyString(req.body.relationshipCategory);
+        enums.categories.get(req.body.relationshipCategory); // will throw an error if the category is not in the list
+    }
+    catch(e) {
+        //res.status(400).render('frames/error', {layout: 'profile', error: e});
+        res.status(400).json({error:e});
+        return;
+    }
+
+    let relationshipObj;
+    try {
+        // Need to add to the userObject
+        relationshipObj = await relationships.createRelationship(req.body.relationshipDescription, mentorId, menteeId, req.body.relationshipCategory);
+        let added = await users.updateUserRelationships(userId, relationshipObj);
+       
+        //res.render('frame/request', {layout: 'profile', relationship: relationshipObj, created: true});
+        res.status(200).json({success: true, relationship: relationshipObj, added: added}); // used for testing
+    }
+    catch(e){
+        //res.status(500).render('frames/error', {layout: 'profile', error: "Internal server error"});
+        res.status(500).json({error:e});
+    }
+};
+
+/**
+ * Get relationshipObjects filtered by status
+ * @param {Object} req - request object
+ * @param {Object} res - response object
+ */
+async function getRelationshipByStatus(req, res){
+    // Show only the relationships in requested status
+    let returnList;
+    let userId;
+
+    try {
+        userId = validate.convertID(req.params.userId);
+        if (!(req.params.status instanceof data.status)) throw `Error: Must provide a status enum`;
+        // TODO: if in mentor view, show all your mentees
+        let menteeList = await users.getMenteeList(userId);

```

Comment:
Performance enhancement for the future! (Optional for now)
In current implementation:
1. Hit a User for retrieving the Mentor List (say size is M)
2. Hit a User for retrieving the Mentee List (say size is N)
3. Hit the Relationships for filtering by status ( = M+N times)
4. Hit the Relationships for getting a Relationship (= M+N times)

I guess We can enhance this to:
1. Retrieve the mentor and mentee lists in one hit
2. Retrieve all the Relationship objects in another hit (Using $In operator of Mongo)
3. Filter them in Route or in data